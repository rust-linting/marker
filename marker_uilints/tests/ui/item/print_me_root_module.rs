//! This is so hacky, and I love it! So what is this test?
//! Well, uilints has a test, which prints items that start with `print_me`,
//! and the root module of a crate has the name of the crate. So, my naming the
//! file `print_me_*` we have created a test case, that prints the root module
//! automatically. :D

fn main() {}
