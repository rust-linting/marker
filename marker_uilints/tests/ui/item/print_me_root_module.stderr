warning: printing item
  |
  = note: Mod(
              ModItem {
                  data: CommonItemData {
                      id: ItemId(..),
                      span: SpanId(..),
                      vis: Visibility {
                          span: None,
                          sem: Visibility {
                              _lifetime: PhantomData<&()>,
                              kind: DefaultPub,
                          },
                      },
                      ident: Ident {
                          name: "print_me_root_module",
                          span: $DIR/print_me_root_module.rs:1:1 - 1:1,
                      },
                  },
                  items: [
                      Fn(
                          FnItem {
                              data: CommonItemData {
                                  id: ItemId(..),
                                  span: SpanId(..),
                                  vis: Visibility {
                                      span: None,
                                      sem: Visibility {
                                          _lifetime: PhantomData<&()>,
                                          kind: Default(
                                              ItemId(..),
                                          ),
                                      },
                                  },
                                  ident: Ident {
                                      name: "main",
                                      span: $DIR/print_me_root_module.rs:7:4 - 7:8,
                                  },
                              },
                              generics: GenericParams {
                                  params: [],
                                  clauses: [],
                              },
                              constness: NotConst,
                              syncness: Sync,
                              safety: Safe,
                              is_extern: false,
                              has_self: false,
                              abi: Default,
                              params: [],
                              return_ty: None,
                              body_id: Some(
                                  BodyId(..),
                              ),
                          },
                      ),
                  ],
              },
          )
  = note: `#[warn(marker::marker_uilints::test_lint)]` on by default

warning: 1 warning emitted

